src/irr/asset/CImageLoaderTGA.cpp

```diff
@@ -112,78 +112,129 @@ bool CImageLoaderTGA::isALoadableFileFormat(io::IReadFile* _file) const
 	return true;
 }
 
-// convertColorFlip() does color conversion as well as taking care of properly flipping the given image.
+/*
+	Targa formats needs two y-axis flips. The first is a flip to get the Y conforms to OpenGL coords.
+	The second flip is defined from within the .tga file itself (header.ImageDescriptor & 0x20).
+	if flip - perform two flips (OpenGL + Targa) = no flipping. Don't flip the image at all in that case
+	if not - do an OpenGL flip
+*/
+
 template <E_FORMAT srcFormat, E_FORMAT destFormat>
-static void convertColorFlip(uint32_t regionBufferRowLenght, VkExtent3D imageExtent, const core::smart_refctd_ptr<ICPUBuffer>& bufferSourceData, core::smart_refctd_ptr<ICPUBuffer>& bufferOutData, bool flip)
+core::smart_refctd_ptr<ICPUImage> createAndconvertImageData(ICPUImage::SCreationParams& imgInfo, core::smart_refctd_ptr<ICPUBuffer>& texelBuffer, core::smart_refctd_dynamic_array<ICPUImage::SBufferCopy>& regions, bool flip)
```

Comment:
why are the old regions going in as a reference?

Reply:
mistake